### PR TITLE
mtgfetch

### PR DESCRIPTION
**Pull request by woppels**
I squashed correctly here. I will close the other PR.

### CODE REVIEW

File: scripts/mtgfetch.coffee [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		  				msg.send card.imageUrl for card in json.cards when card.set isnt 'pPRE' && card.set isnt 'pMGD' 
+		  			catch error
+		  				msg.send "Card #{cardname} not found."
+		  		else
+		  			msg.send "This shouldn't happen..."
+
+  	# Else no apostrophe
+  	else
+  		cardname = '\"' + cardname + '\"'
+  		#msg.send """https://api.magicthegathering.io/v1/cards?name=#{cardname}"""
```
Comment (newtonrd) [inline scripts/mtgfetch.coffee]: Remove this line if you don't need it

File: scripts/mtgfetch.coffee [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	  			numCards = Object.keys(json.cards).length
+
+	  			# Check if there isn't an exact match
+	  			if numCards > 0
+	  				# Return first card that has an imageUrl.
+	  				for card in json.cards
+	  					if card.imageUrl? 
+	  						msg.send card.imageUrl
+	  						break
+
```
Comment (newtonrd) [inline scripts/mtgfetch.coffee]: Remove this space

File: scripts/mtgfetch.coffee [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	  			sameArray = {}
+	  			sameArray[cardname] = 0
+	  			for card in json.cards
+	  				sameArray[card.name]++
+
+	  			# Check if there are multiple cards that do not include promo or gameday
+	  			if(numCards > 1 && sameArray[cardname] != numCards)
+	  				msg.send '''
+	  					Please be more specific:
+	  				'''
```
Comment (newtonrd) [inline scripts/mtgfetch.coffee:45]: Does this need to be on 3 lines?

File: scripts/mtgfetch.coffee [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	  				msg.send card.name for card in json.cards
+	  			# If there is an exact match for a card name
+	  			else if (sameArray[cardname] == numCards)
```
Comment (newtonrd) [inline scripts/mtgfetch.coffee:48]: How does this work? Are you comparing the name of a card to an integer value?

Comment (woppels) [inline scripts/mtgfetch.coffee:48]: This is a mapping value for the cardname. I'm comparing the count of the cardname instance to the total number of cards. 

There are instances such as Languish or Gideon's Phalanx that have multiple printings of the same card. In this case, I need to not show "Please be more specific" if the number cards found is greater than 1 but they are all identically named.

File: scripts/mtgfetch.coffee (hunk 3 above)
Comment (woppels) [inline scripts/mtgfetch.coffee:45]: It does not.

**Comment by lydonb** [comment]
Go ahead and squash them all again @woppels - let me know if you need some help. Also, it looks like you're committing as @woppels and as Michael Esparza. The ME one is probably due to a weird setup in your git config.

File: scripts/mtgfetch.coffee (hunk 4 above)
Comment (newtonrd) [inline scripts/mtgfetch.coffee:48]: This makes sense, thanks

**Comment by newtonrd** [comment]
:shipit:

**Pull request merged**